clients/go/gengo.go

```diff
@@ -323,15 +323,15 @@ func New(ctx context.Context, basePath string, opts ...Option) *WagClient {
 
 // NewFromDiscovery creates a client from the discovery environment variables. This method requires
 // the three env vars: SERVICE_{{.FormattedServiceName}}_HTTP_(HOST/PORT/PROTO) to be set. Otherwise it returns an error.
-func NewFromDiscovery() (*WagClient, error) {
```

Comment:
This change makes me wonder if there's a different default that might be better or safer. Thoughts?

Reply:
So I spent a while Friday experimenting with that. 

And I know it's something you mentioned here:

`
POSSIBLE ANNOYANCE: Doing things this way - where the client default is not our Clever best practice, but you have to call some library whenever you create a wag client to get it set up with tracing - is okay but not ideal, since it makes it easy to lose instrumentation by accident. Maybe there’s some way to have it all - like call some global “set default wag client options” method then whenever clients are constructed, they’ll pull from that first? Seems tricky to make work without having the client depend on e.g. particular versions of github.com/Clever/wag or the library that includes this thing, etc, but maybe.
`

The actual setup isn't "too" ugly and looks like:
```go 
	dacService, err := dacServiceClient.NewFromDiscovery(
		dacServiceClient.WithLogger(logger.NewConcreteLogger("dac-service-client")),
		dacServiceClient.WithInstrumentor(tracing.InstrumentedTransport),
		dacServiceClient.WithExporter(exporter),
	)
```

The trick is what we REALLY want to do is basically be able to pass in a client "package" to a function and then have it call that package's functions with our options. And there just isn't a way to do that in Go. 

```go 
dacService, err := someConfigPackage.AutoInstrument(dacServiceClient);
```

Reply:
Noop exporter instead of stdouttrace as default exporter now. 

Also bumped stretchr/testify because Andrew did in a wagv8 PR recently, and that bumped a few other things. 

I also downgraded some of the otel stuff after testing (to the versions I was originally working with) because otel has made the decision to lock all future versions to go v1.18+ and will be quickly following new go releases and NOT considering that a breaking change as far as semver as "it's not a breaking change in the API". So I can't just get the latest of everything in the v1.x.x.